test/unit-tests/borrowableMarginPool.test.ts

```diff
@@ -0,0 +1,666 @@
+        1637792800,
+        isPut,
+        {
+          from: owner,
+        },
+      )
+
+      // check otoken whitelist
+      await whitelist.whitelistOtoken(otoken2.address)
+
+      await expectRevert(
+        marginPool.borrow(otoken2.address, 1, { from: random }),
+        'MarginPool: Cannot borrow collateral asset of expired oToken',
+      )
+    })
+
+    it('should revert borrowing if using put oToken', async () => {
+      const otoken2 = await Otoken.new()
+
+      const strikePrice = createTokenAmount(200)
+      const isPut = true
+
+      await otoken2.init(
+        addressBook.address,
+        weth.address,
+        usdc.address,
+        usdc.address,
+        strikePrice,
+        1637792800,
+        isPut,
+        {
+          from: owner,
+        },
+      )
+
+      // check otoken whitelist
+      await whitelist.whitelistOtoken(otoken2.address)
+
+      await expectRevert(
+        marginPool.borrow(otoken2.address, 1, { from: random }),
+        'MarginPool: oToken is not a call option',
+      )
+    })
+
+    it('should revert borrowing if using blacklisted oToken', async () => {
+      const otoken2 = await Otoken.new()
+
+      const strikePrice = createTokenAmount(200)
+      const isPut = false
+
+      await otoken2.init(
+        addressBook.address,
+        weth.address,
+        usdc.address,
+        usdc.address,
+        strikePrice,
+        1637792800,
+        isPut,
+        {
+          from: owner,
+        },
+      )
+
+      await expectRevert(
+        marginPool.borrow(otoken2.address, 1, { from: random }),
+        'MarginPool: oToken is not whitelisted',
+      )
+    })
+
+    it('should revert borrowing if borrowing more than allocated', async () => {
+      await expectRevert(
+        marginPool.borrow(otoken.address, 1, { from: random }),
+        'MarginPool: Borrowing more than allocated',
+      )
+
+      await expectRevert(
+        marginPool.borrow(otoken.address, new BigNumber(await otoken.balanceOf(user1)).plus(1), { from: user1 }),
+        'MarginPool: Borrowing more than allocated',
+      )
+    })
+
+    it('should revert borrowing if borrowing more than allocated (with borrowPCT chance)', async () => {
+      const borrowPCT = 5000
+
+      await marginPool.setBorrowPCT(await otoken.collateralAsset(), borrowPCT, { from: owner })
+
+      const oTokenAmount = new BigNumber(await otoken.balanceOf(user1))
+      const oTokenToUnderlying = oTokenAmount.times(new BigNumber(10).exponentiatedBy(10))
+
+      assert.equal(
+        oTokenToUnderlying.div(2).toString(),
+        new BigNumber(await marginPool.borrowable(user1, otoken.address)).toString(),
+        'Borrowable amount mismatch',
+      )
+
+      await expectRevert(
+        marginPool.borrow(otoken.address, oTokenAmount.div(2).plus(1), { from: user1 }),
+        'MarginPool: Borrowing more than allocated',
+      )
+    })
+
+    it('should revert borrowing after borrowing full amount', async () => {
```

Comment:
I think this is testing same as above, not after any borrow of full amount - its just a user with no balance.

Reply:
Yes it is. Will delete.